### PR TITLE
feature/passwordValidation

### PR DESCRIPTION
**Pull request by ZeshanShahid79**

### CODE REVIEW

**Review by Flooooooooooorian — CHANGES_REQUESTED** [review]

File: backend/pom.xml [hunk 1]
```diff
@@ -69,7 +58,7 @@
                 <configuration>
                     <excludes>
                         <exclude>
-                            <groupId>org.projectlombok</groupId>
+                            <groupId>org.project lombok</groupId>
```
Comment (Flooooooooooorian) [inline backend/pom.xml:61]: Why did you add a space here? 🤔

File: backend/src/test/java/de/neuefische/ffmjava221/teamprojekt/backend/guest/GuestIntegrationTest.java [hunk 2]
```diff
@@ -53,12 +54,42 @@ void addGuest() throws Exception {
...
+                                {"firstName": "test",
+                                 "lastName": "test",
+                                 "email": "test@gmail.com",
+                                 "password": "password",
+                                 "confirmPassword": "password"}
+                                """))
+                .andExpect(status().isBadRequest());
+    }
+
+//    @Test
```
Comment (Kreyno93) [inline backend/src/test/java/de/neuefische/ffmjava221/teamprojekt/backend/guest/GuestIntegrationTest.java:78]: remove commented code

File: backend/src/main/java/de/neuefische/ffmjava221/teamprojekt/backend/guest/NewGuest.java [hunk 3]
```diff
@@ -10,6 +11,7 @@ public record NewGuest(
         String lastName,
         @Email
         String email,
+        @Pattern(regexp = "^(?=[^A-Z]*+[A-Z])(?=[^a-z]*+[a-z])(?=\\D*+\\d)(?=[^#?!@$ %^&*-]*+[#?!@$ %^&*-]).{8,}$",message = "Password must have minimum eight characters, at least one letter and one number!")
         String password,
         String confirmPassword
```
Comment (Flooooooooooorian) [inline backend/src/main/java/de/neuefische/ffmjava221/teamprojekt/backend/guest/NewGuest.java:16]: You could remove the confirmPassword...

File: backend/src/test/java/de/neuefische/ffmjava221/teamprojekt/backend/guest/GuestServiceTest.java [hunk 4]
```diff
@@ -102,5 +104,44 @@ void updateGuestByInvalidId() {
             System.out.println(e.getMessage());
         }
     }
+
+//    @Test
+//    void checkConfirmPassword() {
```
Comment (Kreyno93) [inline backend/src/test/java/de/neuefische/ffmjava221/teamprojekt/backend/guest/GuestServiceTest.java:109]: remove commented code

File: backend/src/test/java/de/neuefische/ffmjava221/teamprojekt/backend/guest/GuestIntegrationTest.java [hunk 5]
```diff
@@ -53,12 +54,42 @@ void addGuest() throws Exception {
...
+//                        .content("""
+//                                {"firstName": "test",
+//                                 "lastName": "test",
+//                                 "email": "test@gmail.com",
+//                                 "password": "SuperSecret344$$",
+//                                 "confirmPassword": "SuperSecret344$$"}
+//                                """))
+//                .andExpect(status().isBadRequest());
+//    }
+
```
Comment (Flooooooooooorian) [inline backend/src/test/java/de/neuefische/ffmjava221/teamprojekt/backend/guest/GuestIntegrationTest.java:92]: Remove this

File: backend/src/test/java/de/neuefische/ffmjava221/teamprojekt/backend/guest/GuestServiceTest.java [hunk 6]
```diff
@@ -102,5 +104,44 @@ void updateGuestByInvalidId() {
...
+//        NewGuest guest1 = new NewGuest("Steven", "Lang", "fsagfg@gmail.com", "hallo123", "hallo");
+//
+//        //WHEN
+//
+//        boolean actual = guestService.checkConfirmPassword(guest1);
+//
+//        //THEN
+//
+//        assertEquals(false, actual);
+//    }
```
Comment (Flooooooooooorian) [inline backend/src/test/java/de/neuefische/ffmjava221/teamprojekt/backend/guest/GuestServiceTest.java:145]: Remove pls

File: frontend/src/Guest/RegisterForm.tsx [hunk 7]
```diff
@@ -21,16 +22,25 @@ export default function RegisterForm() {
...
     const [confirmPassword, setConfirmPassword] = useState("");
+    const [error, setError] = useState("");
 
     const handleFormSubmit = (event: ChangeEvent<HTMLFormElement>) => {
         event.preventDefault();
+        if (!isValidEmail(email)) {
+            setError("Email is invalid");
+        } else {
+            setError("");
+        }
```
Comment (Flooooooooooorian) [inline frontend/src/Guest/RegisterForm.tsx:33]: Do you still want to continue sending the form and the registration process even the email is invalid? 🤔

**Review by Kreyno93 — CHANGES_REQUESTED** [review]

File: frontend/src/Guest/RegisterForm.tsx [hunk 8]
```diff
@@ -21,16 +22,25 @@ export default function RegisterForm() {
...
         postForm();
         setFirstName("");
         setLastName("");
         setEmail("");
         setPassword("");
         setConfirmPassword("")
     }
+    const isValidEmail= (email:string) => {
+        return /.@./.test(email);
```
Comment (Flooooooooooorian) [inline frontend/src/Guest/RegisterForm.tsx:42]: Maybe not the best regex for email validation, but checking an email adress is not that easy. 
`abc@abc` would be a a valid email adress, since you check the adress in the backend i would say thats enough for now

**Review by Flooooooooooorian — COMMENTED** [review]
Added some more comments 😄

**Comment by ZeshanShahid79** [comment]
can we bitte mergen jetzt :D

**Review by Flooooooooooorian — DISMISSED** [review]

**Review by Kreyno93 — APPROVED** [review]

**Review by chris-yooo — APPROVED** [review]

**Comment by ZeshanShahid79** [comment]
approved by group 5
